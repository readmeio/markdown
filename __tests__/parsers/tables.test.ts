import { removePosition } from 'unist-util-remove-position';

import { mdast } from '../../index';

describe('table parser', () => {
  describe('unescaping pipes', () => {
    it('parses tables with pipes in inline code', () => {
      const doc = `
|              |    |
| :----------- | :- |
| \`foo \\| bar\` |    |
`;
      const ast = mdast(doc);
      removePosition(ast, { force: true });

      expect(ast).toMatchSnapshot();
    });

    it('parses tables with pipes', () => {
      const doc = `
|            |    |
| :--------- | :- |
| foo \\| bar |    |
`;
      const ast = mdast(doc);
      removePosition(ast, { force: true });

      expect(ast).toMatchSnapshot();
    });

    it('parses jsx tables with pipes in inline code', () => {
      const doc = `
<Table align={["left","left"]}>
  <thead>
    <tr>
      <th style={{ textAlign: "left" }}>
        force
        jsx
      </th>

      <th style={{ textAlign: "left" }}>

      </th>
    </tr>
  </thead>

  <tbody>
    <tr>
      <td style={{ textAlign: "left" }}>
        \`foo | bar\`
      </td>

      <td style={{ textAlign: "left" }}>

      </td>
    </tr>
  </tbody>
</Table>
`;

      const ast = mdast(doc);
      removePosition(ast, { force: true });

      expect(ast).toMatchSnapshot();
    });
  });
});
