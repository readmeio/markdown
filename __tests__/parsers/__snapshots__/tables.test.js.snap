// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Parse magic block tables > renders an table with missing cells 1`] = `
{
  "children": [],
  "type": "root",
}
`;
